TranslucentTB/taskbar/taskbarattributeworker.cpp

```diff
@@ -317,7 +317,15 @@ TaskbarAppearance TaskbarAttributeWorker::GetConfig(taskbar_iterator taskbar) co
 
 	if (m_Config.MaximisedWindowAppearance.Enabled && SetContainsValidWindows(taskbar->second.MaximisedWindows))
 	{
-		return WithPreview(txmp::TaskbarState::MaximisedWindow, m_Config.MaximisedWindowAppearance);
+		const std::optional<Rule> rule = m_Config.MaximisedWindowAppearance.FindRule(window);
+		if (rule)
+		{
+			return WithPreview(txmp::TaskbarState::MaximisedWindow, rule.value());
+		}
+		else
+		{
+			return WithPreview(txmp::TaskbarState::MaximisedWindow, m_Config.MaximisedWindowAppearance);
+		}
```

Comment:
I am currently stuck here with this error 
```
Build started...
1>------ Build started: Project: TranslucentTB, Configuration: Debug x64 ------
1>taskbarattributeworker.cpp
1>D:\dev\TranslucentTB\TranslucentTB\taskbar\taskbarattributeworker.cpp(320,86): error C2662: 'const std::optional<Rule> RuledTaskbarAppearance::FindRule(Window)': cannot convert 'this' pointer from 'const RuledTaskbarAppearance' to 'RuledTaskbarAppearance &'
1>D:\dev\TranslucentTB\TranslucentTB\taskbar\taskbarattributeworker.cpp(320,86): error C2662: 		const std::optional<Rule> rule = m_Config.MaximisedWindowAppearance.FindRule(window);
1>D:\dev\TranslucentTB\TranslucentTB\taskbar\taskbarattributeworker.cpp(320,86): error C2662: 		                                                                                   ^
1>D:\dev\TranslucentTB\TranslucentTB\taskbar\taskbarattributeworker.cpp(320,44): message : Conversion loses qualifiers
1>D:\dev\TranslucentTB\TranslucentTB\taskbar\taskbarattributeworker.cpp(320,44): message : 		const std::optional<Rule> rule = m_Config.MaximisedWindowAppearance.FindRule(window);
1>D:\dev\TranslucentTB\TranslucentTB\taskbar\taskbarattributeworker.cpp(320,44): message : 		                                         ^
1>D:\dev\TranslucentTB\Common\config\ruledtaskbarappearance.hpp(46,35): message : see declaration of 'RuledTaskbarAppearance::FindRule'
1>D:\dev\TranslucentTB\Common\config\ruledtaskbarappearance.hpp(46,35): message : 	inline const std::optional<Rule> FindRule(Window window)
1>D:\dev\TranslucentTB\Common\config\ruledtaskbarappearance.hpp(46,35): message : 	                                 ^
1>Done building project "TranslucentTB.vcxproj" -- FAILED.
2>------ Build started: Project: AppPackage, Configuration: Debug x64 ------
2>Xaml.vcxproj -> D:\dev\TranslucentTB\x64\Debug\Xaml\Xaml.dll
2>C:\Program Files\Microsoft Visual Studio\2022\Preview\MSBuild\Microsoft\VisualStudio\v17.0\AppxPackage\Microsoft.AppXPackage.Targets(3364,5): warning : APPX4001: Build property AppxBundlePlatforms is not explicitly set and is calculated based on currently building architecture. Use 'Create App Package' wizard or edit project file to set it.
2>Done building project "Xaml.vcxproj".
========== Build: 1 succeeded, 1 failed, 3 up-to-date, 0 skipped ==========
========== Deploy: 0 succeeded, 0 failed, 0 skipped ==========

```

Reply:
It's because FindRule is not a const member. Setting it to be const should solve the issue.

Reply:
thanks, my cpp skills are shit, and I was trying to remove/add const from the return type and not from the function itself.